plugins/auth/index.js

```diff
@@ -140,30 +140,58 @@ exports.register = (server, options, next) => {
+                    const userFactory = request.server.app.userFactory;
+                    const pipelineFactory = request.server.app.pipelineFactory;
+
+                    tokenFactory.get({ value: tokenValue })
+                        .then((token) => {
+                            if (!token) {
+                                return Promise.reject();
+                            } else if (token.userId) {
+                                // if token has userId then the token is for user
+                                return userFactory.get(token.userId)
+                                    .then((user) => {
+                                        if (!user) {
+                                            return Promise.reject();
+                                        }
+
+                                        return {
+                                            username: user.username,
+                                            scmContext: user.scmContext,
+                                            scope: ['user']
+                                        };
+                                    });
+                            } else if (token.pipelineId) {
+                                // if token has pipelineId then the token is for pipeline
+                                return pipelineFactory.get(token.pipelineId)
+                                    .then((pipeline) => {
+                                        if (!pipeline) {
+                                            return Promise.reject();
+                                        }
+
+                                        return pipeline.admin.then(admin => ({
+                                            username: admin.username,
+                                            scmContext: pipeline.scmContext,
+                                            pipelineId: token.pipelineId,
+                                            scope: ['pipeline']
+                                        }));
+                                    });
                             }
 
-                            request.log(['auth'], `${user.username} has logged in via API keys`);
-
-                            const profile = {
-                                username: user.username,
-                                scmContext: user.scmContext,
-                                scope: ['user']
-                            };
-
+                            return Promise.reject();
+                        })
+                        .then((profile) => {
+                            request.log(['auth'], `${profile.username} has logged in via API keys`);
                             profile.token = server.plugins.auth.generateToken(profile);
 
                             return cb(null, true, profile);
-                        });
+                        })
+                        .catch(() => cb(null, false, {}));
```

Comment:
For the purpose of debugging, shall we cat the err and log it out?

Reply:
Sure, I'll fix it.